Move default Run 4 workflows to D121

kpedro88 opened the pull request:
#### PR description:

This PR is intended to complete the update of the default Run 4 geometry to D121, started in #48297.

#### PR validation:

runTheMatrix.py lists the correct workflows.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport. Can be backported upon request, but no immediate needs are foreseen.

kpedro88 committed:
update relvals to use D121

kpedro88 committed:
update CondTools/Geometry to use D121

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49437/46917

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #49419
  - File  Configuration/PyReleaseValidation/scripts/runTheMatrix.py  modified in PR(s): #48637, #49419

cmsbuild commented:
A new Pull Request was created by  @kpedro88 for master.

It involves the following packages:

- CondTools/Geometry (**db**)
- Configuration/PyReleaseValidation (**pdmv**)
- Configuration/StandardSequences (**operations**)


@AdrianoDee, @Alejandro1400, @DickyChant, @JanChyczynski, @antoniovagnerini, @arunhep, @atpathak, @cmsbuild, @davidlange6, @fabiocos, @francescobrivio, @ftenchini, @mandrenguyen, @miquork, @perrotta can you please review it and eventually sign? Thanks.
@GiacomoSguazzoni, @JanChyczynski, @Martin-Grunewald, @PonIlya, @VinInn, @VourMa, @bsunanda, @dgulhan, @elusian, @fabiocos, @felicepantaleo, @makortel, @mmasciov, @mmusich, @mtosi, @rovere, @rsreds, @sameasy, @slomeo, @yuanchao this is something you requested to watch as well.
@ftenchini, @mandrenguyen, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

kpedro88 commented:
Open questions:
1. @cms-sw/geometry-l2 I updated the CondTools scripts as we have done before. There are many scripts in `Geometry/[Package]/test/python` that also use D110. I am not sure which of these are actively used, so I have not addressed them. Please let me know if there are any that are particularly important to update.
2. @cms-sw/hlt-l2 I updated the `ph2_hlt` section of the matrix to D121 as well. Please let me know if this is not desired.

kpedro88 commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -4768,6 +4768,7 @@ def gen2024HiMix(fragment,howMuch):
 defaultDataSets['Run4D96']='CMSSW_13_1_0_pre1-130X_mcRun4_realistic_v2_2026D96noPU-v'
 defaultDataSets['Run4D98']='CMSSW_13_2_0_pre1-131X_mcRun4_realistic_v5_2026D98noPU-v'
 defaultDataSets['Run4D110']='CMSSW_15_1_0_pre5-150X_mcRun4_realistic_v1_STD_RegeneratedGS_Run4D110_noPU-v'
+defaultDataSets['Run4D121']=''
```
@cms-sw/pdmv-l2 is there anything that we can put here? I could not find any D121 samples in DAS.

AdrianoDee replied:
Not really. We can produce them. 

In general I think that if we want to move definetly to D121, we'd need a validation campaign.

srimanob replied:
Maybe @AdrianoDee could help to submit minbias relvals with D121 with existing release (e.g. 16_0_0_pre1 or pre2). Then, within the next few days, when dataset is available, this PR can be updated.

srimanob replied:
By the way, not sure how short matrix for PR test will work for now, as it will try to run D121+PU workflow, right?

kpedro88 replied:
Including pre2 relvals for this geometry in this PR would be ideal.

Indeed, a few tests may fail, but I think most do not include PU.

AdrianoDee replied:
I've sent a set of D121 relvals [here](https://cms-pdmv-prod.web.cern.ch/relval/relvals?ticket=CMSSW_16_0_0_pre2__Phase2_Run4D121_noPU_Reference-00001&shown=2099199&page=0&limit=50). Most of them are already done. So there you can use `CMSSW_16_0_0_pre2-150X_mcRun4_realistic_v1_STD_RegeneratedGS_Run4D121_noPU-v`.

For D121 validation itself, since it should be minor, I would proceed by just having both D110 and D121 samples produced for the incoming `16_0_0_pre3` campaign and have the validators inspect them there.

kpedro88 replied:
Thanks @AdrianoDee ! Your proposal for validation sounds good to me.

mmusich commented:
test parameters:
   * workflows = ph2_hlt

mmusich commented:
> I updated the ph2_hlt section of the matrix to D121 as well. Please let me know if this is not desired.

this is going to be discussed this week at the upgrade HLT meeting.

kpedro88 committed:
add D121 relval string

kpedro88 commented:
Thanks @mmusich. We can wait to merge this PR until we get the decision from HLT.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49437/46933

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #49419
  - File  Configuration/PyReleaseValidation/scripts/runTheMatrix.py  modified in PR(s): #48637, #49419

kpedro88 commented:
@mmusich done in fd7cf9e72f21423c5816b089bcef694146b4e4d7 (in such a way that later updates should be easy, i.e. just change the one line to be `prefixDetHLT = prefixDet`)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49437/46934

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/scripts/runTheMatrix.py  modified in PR(s): #48637

cmsbuild commented:
Pull request #49437 was updated. @AdrianoDee, @Alejandro1400, @DickyChant, @JanChyczynski, @antoniovagnerini, @arunhep, @atpathak, @cmsbuild, @davidlange6, @fabiocos, @francescobrivio, @ftenchini, @mandrenguyen, @miquork, @perrotta can you please check and sign again.

mandrenguyen commented:
> If possible, this new default should enter 16_0_X. (Hopefully along with some cleanup of old geometries, which is the next step, but can be pursued in parallel). Questions:
> 
> 1. Will the validation converge before the pre4 deadline?
> 2. @cms-sw/orp-l2 is pre4 still open for this kind of change?

At least since the time I started this job, it seems that we have not enforced the "no new features for last pre-release in cycle" rule that I remember from the past.  But policy aside, we should come to an agreement with yourself, Adriano, et al, whether it makes sense to squeeze this in pre3 or not.

makortel commented:
> it seems that we have not enforced the "no new features for last pre-release in cycle" rule that I remember from the past.

Just to clarify, for 16_0_X, would that mean "no new features after pre3" or "no new features after pre4"?

kpedro88 commented:
please test

kpedro88 committed:
update era for D121

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49437/46988

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_steps.py  modified in PR(s): #49462
  - File  Configuration/PyReleaseValidation/scripts/runTheMatrix.py  modified in PR(s): #48637

kpedro88 commented:
please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1d2dd6/49702/summary.html
**COMMIT**: 893cc1c635d0bd251f01d89c94945409ede585d7
**CMSSW**: CMSSW_16_0_X_2025-11-26-1100/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/49437/49702/install.sh` to create a dev area with all the needed externals and cmssw changes.

**DAS Queries**: The DAS query tests failed, see the summary page for details.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 6 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-26-1100+1d2dd6/72060/validateJR.html)
 - Reco comparison had  4 failed jobs
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3554205
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-26-1100+1d2dd6/72060/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3554182
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 46 files compared)
 - Checked 199 log files, 175 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-11-26-1100+1d2dd6/72060/triggerResults): no differences found

perrotta commented:
+db

mandrenguyen commented:
+1
